pyro/primitives.py

```diff
@@ -6,18 +6,22 @@
 from collections import OrderedDict
 from contextlib import ExitStack, contextmanager
 from inspect import isclass
+from typing import Callable, Dict, Iterator, Optional, Sequence, Union
 
 import torch
+from torch.distributions import constraints
```

Comment:
I see lots of new dependencies here.  Historically we've had a number of cyclic dependency issues in Pyro.  One thing we might consider to try to avoid cyclid dependencies is to guard these with an `if TYPE_CHECKING`:
```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from torch.distributions import contstraints
    from pyro.distributions import TorchDistribution
    from pyro.params.param_store import ParamStoreDict
    from pyro.poutine.runtime import Message
```
Or maybe we can just use that trick next time we need to fix a cyclic dependency. Either way.

Reply:
Great idea. However, in this case it turns out that `make docs` fails without `TorchDistribution` and `ParamStoreDict` imported.  And `constraints` is used in actual code.